h2/config.py

```diff
@@ -6,6 +6,27 @@
 Objects for controlling the configuration of the HTTP/2 stack.
 """
 
+import weakref
+
+
+class _BooleanConfigOption(object):
+    """
+    Descriptor for handling a boolean config option.  This will block
+    attempts to set boolean config options to non-bools.
+    """
+    def __init__(self, name, default):
+        self.name = name
+        self.default = default
+        self.data = weakref.WeakKeyDictionary()
```

Comment:
Is it really necessary to use a weakref here? This is doubly true given that we're only storing bools, which are definitionally singletons and so cannot possibly cause us trouble to just store directly.